File: BleModule/src/main/java/com/topdon/ble/ScanConfiguration.java
package com.topdon.ble;

import android.bluetooth.le.ScanFilter;
import android.bluetooth.le.ScanSettings;
import android.os.Build;

import androidx.annotation.RequiresApi;

import java.util.List;

public class ScanConfiguration {
    int scanPeriodMillis = 10000;
    boolean acceptSysConnectedDevice;
    ScanSettings scanSettings;
    boolean onlyAcceptBleDevice;
    int rssiLowLimit = -120;
    List<ScanFilter> filters;

    public int getScanPeriodMillis() {
        return scanPeriodMillis;
    }

    public ScanConfiguration setScanPeriodMillis(int scanPeriodMillis) {
        //1
        if (scanPeriodMillis >= 1000) {
            this.scanPeriodMillis = scanPeriodMillis;
        }
        return this;
    }

    public boolean isAcceptSysConnectedDevice() {
        return acceptSysConnectedDevice;
    }

    public ScanConfiguration setAcceptSysConnectedDevice(boolean acceptSysConnectedDevice) {
        this.acceptSysConnectedDevice = acceptSysConnectedDevice;
        return this;
    }

    public ScanSettings getScanSettings() {
        return scanSettings;
    }

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    public ScanConfiguration setScanSettings(ScanSettings scanSettings) {
        Inspector.requireNonNull(scanSettings, "scanSettings can't be null");
        this.scanSettings = scanSettings;
        return this;
    }

    public boolean isOnlyAcceptBleDevice() {
        return onlyAcceptBleDevice;
    }

    public ScanConfiguration setOnlyAcceptBleDevice(boolean onlyAcceptBleDevice) {
        this.onlyAcceptBleDevice = onlyAcceptBleDevice;
        return this;
    }

    public int getRssiLowLimit() {
        return rssiLowLimit;
    }

    public ScanConfiguration setRssiLowLimit(int rssiLowLimit) {
        this.rssiLowLimit = rssiLowLimit;
        return this;
    }

    public List<ScanFilter> getFilters() {
        return filters;
    }

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    public ScanConfiguration setFilters(List<ScanFilter> filters) {
        this.filters = filters;
        return this;
    }
}

File: BleModule/src/main/java/com/mpdc4gsr/ble/ScanConfiguration.kt
package com.mpdc4gsr.ble

import android.bluetooth.le.ScanFilter
import android.bluetooth.le.ScanSettings
import android.os.Build
import androidx.annotation.RequiresApi

class ScanConfiguration {
    var scanPeriodMillis: Int = 10000
    var isAcceptSysConnectedDevice: Boolean = false
    var scanSettings: ScanSettings? = null
    var isOnlyAcceptBleDevice: Boolean = false
    var rssiLowLimit: Int = -120
    var filters: MutableList<ScanFilter?>? = null

    fun setScanPeriodMillis(scanPeriodMillis: Int): ScanConfiguration {
        if (scanPeriodMillis >= 1000) {
            this.scanPeriodMillis = scanPeriodMillis
        }
        return this
    }

    fun setAcceptSysConnectedDevice(acceptSysConnectedDevice: Boolean): ScanConfiguration {
        this.isAcceptSysConnectedDevice = acceptSysConnectedDevice
        return this
    }

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    fun setScanSettings(scanSettings: ScanSettings?): ScanConfiguration {
        Inspector.requireNonNull<ScanSettings?>(scanSettings, "scanSettings can't be null")
        this.scanSettings = scanSettings
        return this
    }

    fun setOnlyAcceptBleDevice(onlyAcceptBleDevice: Boolean): ScanConfiguration {
        this.isOnlyAcceptBleDevice = onlyAcceptBleDevice
        return this
    }

    fun setRssiLowLimit(rssiLowLimit: Int): ScanConfiguration {
        this.rssiLowLimit = rssiLowLimit
        return this
    }

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    fun setFilters(filters: MutableList<ScanFilter?>?): ScanConfiguration {
        this.filters = filters
        return this
    }
}
